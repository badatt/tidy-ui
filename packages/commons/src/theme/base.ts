// screens https://css-tricks.com/snippets/css/media-queries-for-standard-devices/

const defaultBase = {
  font: {
    bold: 700,
    family:
      'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Ubuntu, "Helvetica Neue", Oxygen, Cantarell, sans-serif`',
    light: 200,
    medium: 400,
    mono: 'ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace',
    regular: 500,
    size: '1rem',
  },
  layout: {
    radius: '0.5rem',
    screens: {
      lg: {
        breakpoint: 1200,
      },
      md: {
        breakpoint: 992,
      },
      sm: {
        breakpoint: 768,
      },
      xl: {
        breakpoint: 1400,
      },
      xs: {
        breakpoint: 576,
      },
    },
    shadow:
      'rgb(255, 255, 255) 0px 0px 0px 0px, rgba(15, 23, 42, 0.05) 0px 0px 0px 1px, rgba(0, 0, 0, 0.1) 0px 20px 25px -5px, rgba(0, 0, 0, 0.1) 0px 8px 10px -6px',
  },
  typography: {
    body1: {
      fontSize: '1rem',
      fontWeight: 400,
      letterSpacing: '0.00938em',
      lineHeight: 1.5,
      marginBottom: '0.5rem',
    },
    body2: {
      fontSize: '0.875rem',
      fontWeight: 400,
      letterSpacing: '0.01071em',
      lineHeight: 1.43,
      marginBottom: '0.4375rem',
    },
    caption: {
      fontSize: '0.75rem',
      fontWeight: 400,
      letterSpacing: '0.03333em',
      lineHeight: 1.66,
      marginBottom: '0.375rem',
    },
    h1: {
      fontSize: '6rem',
      fontWeight: 300,
      letterSpacing: '-0.01562em',
      lineHeight: 1.167,
      marginBottom: '3rem',
    },
    h2: {
      fontSize: '3.75rem',
      fontWeight: 300,
      letterSpacing: '-0.00833em',
      lineHeight: 1.2,
      marginBottom: '1.875rem',
    },
    h3: {
      fontSize: '3rem',
      fontWeight: 400,
      letterSpacing: '0em',
      lineHeight: 1.167,
      marginBottom: '1.5rem',
    },
    h4: {
      fontSize: '2.125rem',
      fontWeight: 400,
      letterSpacing: '0.00735em',
      lineHeight: 1.235,
      marginBottom: '1.0625rem',
    },
    h5: {
      fontSize: '1.5rem',
      fontWeight: 400,
      letterSpacing: '0em',
      lineHeight: 1.334,
      marginBottom: '0.75rem',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      letterSpacing: '0.0075em',
      lineHeight: 1.6,
      marginBottom: '0.325rem',
    },
    hero: {
      fontSize: '6rem',
      fontWeight: 300,
      letterSpacing: '-0.01562em',
      lineHeight: 1.167,
      marginBottom: '3rem',
    },
    p: {
      fontSize: '1rem',
      fontWeight: 400,
      letterSpacing: '0.00938em',
      lineHeight: 1.5,
      marginBottom: '0.5rem',
    },
    span: {
      fontSize: '0.875rem',
      fontWeight: 400,
      letterSpacing: '0.01071em',
      lineHeight: 1.43,
      marginBottom: '0.4375rem',
    },
    subtitle1: {
      fontSize: '2.125rem',
      fontWeight: 400,
      letterSpacing: '0.00735em',
      lineHeight: 1.235,
      marginBottom: '1.0625rem',
    },
    subtitle2: {
      fontSize: '1.5rem',
      fontWeight: 400,
      letterSpacing: '0em',
      lineHeight: 1.334,
      marginBottom: '0.75rem',
    },
    title1: {
      fontSize: '3.75rem',
      fontWeight: 300,
      letterSpacing: '-0.00833em',
      lineHeight: 1.2,
      marginBottom: '1.875rem',
    },
    title2: {
      fontSize: '3rem',
      fontWeight: 400,
      letterSpacing: '0em',
      lineHeight: 1.167,
      marginBottom: '1.5rem',
    },
  },
};

export { defaultBase };
