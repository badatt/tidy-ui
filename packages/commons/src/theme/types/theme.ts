import { DefaultTheme } from 'styled-components';
import { IFont, ILayout, ITypography } from './base';

/**
 * GlobalFont
 */
export interface IGlobalFont {
  /**
   * css `font-family` equivalent
   */
  family?: string;

  /**
   * when given, added to `@import url()` in css
   */
  url?: string;
}

/**
 * GlobalDefault
 */
export interface IGlobalDefault {
  /**
   * theme
   */
  theme?: ITidyUITheme;
}

/**
 * Contract to define a custom theme. You should define all the required properties
 * under this interface to build a custom theme
 *
 * @typedef {ITidyUITheme}
 * @interface ITidyUITheme
 * @augments {DefaultTheme}
 * @public
 */
export interface ITidyUITheme extends DefaultTheme {
  /**
   * Global font settings
   *
   * @type {IFont}
   */
  font: IFont;

  /**
   * Whether the theme is dark
   *
   * @type {boolean}
   */
  isDark: boolean;

  /**
   * Layout configuration
   *
   * @type {ILayout}
   */
  layout: ILayout;

  /**
   * An identifiable & meaningful name to the theme
   *
   * @type {string}
   */
  name: string;

  /**
   * Palette design that holds all color variants in this theme
   *
   * @type {IPalette}
   */
  palette: IPalette;

  /**
   * Typography configuration
   *
   * @type {ITypography}
   */
  typography: ITypography;
}

/**
 * Palette including all color variants
 *
 * @typedef {IPalette}
 * @interface IPalette
 * @public
 */
export interface IPalette {
  /**
   * Component level background colors
   *
   * @type {IBackgroundColor}
   */
  background: IBackgroundColor;

  /**
   * Represents `failure`, `fail`, `error`, `negative`, `sad` state context
   *
   * @type {IColor}
   */
  danger: IColor;

  /**
   * Color used for dividers
   *
   * @type {string}
   */
  divider: string;

  /**
   * Represents `informational` state context
   *
   * @type {IColor}
   */
  info: IColor;

  /**
   * The most significant color that you want to paint in a theme
   *
   * @type {IColor}
   */
  major: IColor;

  /**
   * The second most significant color that you want to paint in a theme
   *
   * @type {IColor}
   */
  minor: IColor;

  /**
   * Represents `neutral`, `default` state context
   *
   * @type {IColor}
   */
  neutral: IColor;

  /**
   * Represents `success`, `pass`, `positive`, `happy` state context
   *
   * @type {IColor}
   */
  success: IColor;

  /**
   * Component level text colors
   *
   * @type {ITextColor}
   */
  text: ITextColor;

  /**
   * Represents `warn`, `alarm` state context
   *
   * @type {IColor}
   */
  warning: IColor;
}

/**
 * Background colors for various components
 *
 * @typedef {IBackgroundColor}
 * @interface IBackgroundColor
 * @public
 */
export interface IBackgroundColor {
  /**
   * Background color for `Card` component
   *
   * @type {string}
   */
  card: string;

  /**
   * Default color for other backgrounds
   *
   * @type {string}
   */
  default: string;

  /**
   * Background color for `Paper` component
   *
   * @type {string}
   */
  paper: string;
}

/**
 * Text colors for various components
 *
 * @typedef {ITextColor}
 * @interface ITextColor
 * @public
 */
export interface ITextColor {
  /**
   * Color to represent disabled text
   *
   * @type {string}
   */
  disabled: string;

  /**
   * Primary text color
   *
   * @type {string}
   */
  primary: string;

  /**
   * Secondary text color
   *
   * @type {string}
   */
  secondary: string;
}

/**
 * Color gradient with various shades
 *
 * @typedef {IColor}
 * @interface IColor
 * @public
 */
export interface IColor {
  /**
   * shade with 50 units
   */
  50: string;
  /**
   * shade with 100 units
   */
  100: string;
  /**
   * shade with 200 units
   */
  200: string;
  /**
   * shade with 300 units
   */
  300: string;
  /**
   * shade with 400 units
   */
  400: string;
  /**
   * shade with 500 units
   */
  500: string;
  /**
   * shade with 600 units
   */
  600: string;
  /**
   * shade with 700 units
   */
  700: string;
  /**
   * shade with 800 units
   */
  800: string;
  /**
   * shade with 900 units
   */
  900: string;
  /**
   * shade with 950 units
   */
  950: string;
  /**
   * All shades
   */
  shades: Record<number, IHsla>;
}

/**
 * Color contract for a hsla color representation
 *
 * @typedef {IHsla}
 * @interface IHsla
 * @public
 */
export interface IHsla {
  /**
   * Alpha (lightness) attribute of the color
   */
  alpha: number;
  /**
   * Hue attribute
   */
  hue: number;
  /**
   * Luminosity attribute
   */
  luminosity: number;
  /**
   * Saturation attribute
   */
  saturation: number;
}
