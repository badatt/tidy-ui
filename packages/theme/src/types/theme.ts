import { ITidyUIBaseTheme } from './base';

export interface ITidyUITheme extends ITidyUIBaseTheme {
  name: string;
  palette: IPalette;
}

interface IPalette {
  major: IColor;
  minor: IColor;
  neutral: IColor;
  info: IColor;
  success: IColor;
  warning: IColor;
  danger: IColor;
  text: ITextColor;
  background: IBackgroundColor;
  divider: string;
}

interface IBackgroundColor {
  paper: string;
  card: string;
  default: string;
}

interface ITextColor {
  primary: string;
  secondary: string;
  disabled: string;
}

export interface IColor {
  50: string;
  100: string;
  200: string;
  300: string;
  400: string;
  500: string;
  600: string;
  700: string;
  800: string;
  900: string;
}

export interface IHsla {
  hue: number;
  saturation: number;
  luminosity: number;
  alpha: number;
}
