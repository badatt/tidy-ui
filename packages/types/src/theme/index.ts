import { ITidyUIBaseTheme as IITidyUIBaseTheme } from './base';
export * from './base';

/**
 * Base theme contract
 *
 * @typedef {ITidyUIBaseTheme}
 * @public
 */
export type ITidyUIBaseTheme = IITidyUIBaseTheme;

import { ITidyUITheme as IITidyUITheme } from './theme';
export * from './theme';

/**
 * Theme contract
 *
 * @typedef {ITidyUITheme}
 * @public
 */
export type ITidyUITheme = IITidyUITheme;
