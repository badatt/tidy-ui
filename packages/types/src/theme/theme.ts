import { ITidyUIBaseTheme } from './base';

/**
 * Contract to define a custom theme. You should define all the required properties
 * under this interface to build a custom theme
 *
 * @public
 * @interface ITidyUITheme
 * @typedef {ITidyUITheme}
 * @augments {ITidyUIBaseTheme}
 */
export interface ITidyUITheme extends ITidyUIBaseTheme {
  /**
   * Whether the theme is dark
   *
   * @type {boolean}
   */
  isDark: boolean;

  /**
   * An identifiable & meaningful name to the theme
   *
   * @type {string}
   */
  name: string;

  /**
   * Palette design that holds all color variants in this theme
   *
   * @type {IPalette}
   */
  palette: IPalette;
}

/**
 * Palette including all color variants
 *
 * @public
 * @interface IPalette
 * @typedef {IPalette}
 */
export interface IPalette {
  /**
   * Component level background colors
   *
   * @type {IBackgroundColor}
   */
  background: IBackgroundColor;

  /**
   * Represents `failure`, `fail`, `error`, `negative`, `sad` state context
   *
   * @type {IColor}
   */
  danger: IColor;

  /**
   * Color used for dividers
   *
   * @type {string}
   */
  divider: string;

  /**
   * Represents `informational` state context
   *
   * @type {IColor}
   */
  info: IColor;

  /**
   * The most significant color that you want to paint in a theme
   *
   * @type {IColor}
   */
  major: IColor;

  /**
   * The second most significant color that you want to paint in a theme
   *
   * @type {IColor}
   */
  minor: IColor;

  /**
   * Represents `neutral`, `default` state context
   *
   * @type {IColor}
   */
  neutral: IColor;

  /**
   * Represents `success`, `pass`, `positive`, `happy` state context
   *
   * @type {IColor}
   */
  success: IColor;

  /**
   * Component level text colors
   *
   * @type {ITextColor}
   */
  text: ITextColor;

  /**
   * Represents `warn`, `alarm` state context
   *
   * @type {IColor}
   */
  warning: IColor;
}

/**
 * Background colors for various components
 *
 * @public
 * @interface IBackgroundColor
 * @typedef {IBackgroundColor}
 */
export interface IBackgroundColor {
  /**
   * Background color for `Card` component
   *
   * @type {string}
   */
  card: string;

  /**
   * Default color for other backgrounds
   *
   * @type {string}
   */
  default: string;

  /**
   * Background color for `Paper` component
   *
   * @type {string}
   */
  paper: string;
}

/**
 * Text colors for various components
 *
 * @public
 * @interface ITextColor
 * @typedef {ITextColor}
 */
export interface ITextColor {
  /**
   * Color to represent disabled text
   *
   * @type {string}
   */
  disabled: string;

  /**
   * Primary text color
   *
   * @type {string}
   */
  primary: string;

  /**
   * Secondary text color
   *
   * @type {string}
   */
  secondary: string;
}

/**
 * Color gradient with various shades
 *
 * @public
 * @interface IColor
 * @typedef {IColor}
 */
export interface IColor {
  50: string;
  100: string;
  200: string;
  300: string;
  400: string;
  500: string;
  600: string;
  700: string;
  800: string;
  900: string;
  shades: Record<number, IHsla>;
}

/**
 * Color contract for a hsla color representation
 *
 * @public
 * @interface IHsla
 * @typedef {IHsla}
 */
export interface IHsla {
  alpha: number;
  hue: number;
  luminosity: number;
  saturation: number;
}
