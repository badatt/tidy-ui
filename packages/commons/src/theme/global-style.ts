import { createGlobalStyle } from 'styled-components';
import { IGlobalStyle } from '@tidy-ui/types';
import { css } from '.';

const GlobalStyle = createGlobalStyle<IGlobalStyle>`
  ${({ theme: { palette }, fontUrl, fontFamily }) => css`
    ${fontUrl
      ? css`
          @import url(${`'${fontUrl}'`});
        `
      : css`
          @import url('https://fonts.googleapis.com/css2?family=PT+Sans:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');
        `}

    *,
    *:before,
    *:after {
      margin: 0;
      padding: 0;
      border: 0;
      font-size: 100%;
      font: inherit;
      vertical-align: baseline;
      box-sizing: border-box;
    }

    *[hidden] {
      display: none;
    }

    html {
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      font-size: 16px;
    }

    body {
      color: ${palette.text.primary};
      min-height: 100vh;
      height: 100%;
      letter-spacing: 0.025rem;
      line-height: 1;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      text-rendering: optimizeLegibility;
      font-family: ${fontFamily ||
      `'PT Sans', 'Helvetica Neue', Arial, 'PingFang SC', 'microsoft yahei', 'hiragino sans gb',
        '游ゴシック Medium', 'Yu Gothic Medium', YuGothicM, 游ゴシック体, YuGothic, 'ヒラギノ角ゴ Pro',
        'Hiragino Kaku Gothic Pro', 'Meiryo UI', メイリオ, Meiryom, 'PT Sans', 'Apple SD Gothic Neo', 'Malgun Gothic',
        Dotum, 'Noto Sans CJK KR', sans-serif;`};
    }

    article,
    aside,
    details,
    figcaption,
    figure,
    footer,
    header,
    hgroup,
    menu,
    nav,
    section {
      display: block;
    }

    menu,
    ol,
    ul {
      list-style: none;
    }
    blockquote,
    q {
      quotes: none;
    }
    blockquote:before,
    blockquote:after,
    q:before,
    q:after {
      content: '';
      content: none;
    }
    table {
      border-collapse: collapse;
      border-spacing: 0;
    }

    img {
      height: auto;
      max-width: 100%;
    }

    a {
      text-decoration: none;
      color: inherit;
    }

    button {
      appearance: none;
      background-color: transparent;
      cursor: pointer;
      display: inline-block;
      line-height: 1;
    }
  `}
`;

export { GlobalStyle };
