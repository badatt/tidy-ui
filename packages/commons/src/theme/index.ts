import baseStyled, { css as baseCss, ThemedCssFunction, ThemedStyledInterface } from 'styled-components';

export { color } from './colors';
export * from './utils';

import { ITidyUITheme } from './types';
export * from './types';
const styled = baseStyled as ThemedStyledInterface<ITidyUITheme>;
const css = baseCss as ThemedCssFunction<ITidyUITheme>;

export { css, styled };

export { defaultBase as baseTheme } from './base';
export { theme as orchidDark } from './orchid-dark';
export { theme as orchidLight } from './orchid-light';
