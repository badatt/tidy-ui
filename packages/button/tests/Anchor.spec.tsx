import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import { orchidLight, orchidDark } from '@tidy-ui/theme';
import { Anchor } from '..';

describe('Render Anchor', () => {
  it('Renders with no styles', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidLight}>
          <Anchor href="https://www.google.com/">simple link</Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders with no styles dark', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidDark}>
          <Anchor href="https://www.google.com/">simple link</Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('External anchor link', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidLight}>
          <Anchor href="https://www.google.com/" newTab>
            external link
          </Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('External anchor link dark', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidDark}>
          <Anchor href="https://www.google.com/" newTab>
            external link
          </Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Disabled anchor link', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidLight}>
          <Anchor href="https://www.google.com/" disable>
            external link
          </Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Disabled anchor link dark', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={orchidDark}>
          <Anchor href="https://www.google.com/" disable>
            external link
          </Anchor>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
