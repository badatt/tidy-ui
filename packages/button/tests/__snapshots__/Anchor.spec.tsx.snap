// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Anchor Disabled anchor link 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_top"
>
  external link
</a>
`;

exports[`Render Anchor Disabled anchor link dark 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_top"
>
  external link
</a>
`;

exports[`Render Anchor External anchor link 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_blank"
>
  external link
  <i
    className=""
    href="https://www.google.com/"
    size="sm"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable={false}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
  </i>
</a>
`;

exports[`Render Anchor External anchor link dark 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_blank"
>
  external link
  <i
    className=""
    href="https://www.google.com/"
    size="sm"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable={false}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
  </i>
</a>
`;

exports[`Render Anchor Renders with no styles 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_top"
>
  simple link
</a>
`;

exports[`Render Anchor Renders with no styles dark 1`] = `
<a
  className=""
  href="https://www.google.com/"
  role="anchor"
  size="sm"
  target="_top"
>
  simple link
</a>
`;
